src/service.rs

```diff
@@ -619,6 +642,18 @@ where
             };
 
             tracing::info!("Got vid commitment for block {id}",);
+
+            // This block was truncated, but we got VID in time.
+            // Maybe we can handle bigger blocks?
+            if block_info.truncated {
+                // Increase max block size by 10%
+                let mut global_state = self.global_state.write_arc().await;
+                global_state.max_block_size = std::cmp::min(
+                    global_state.max_block_size + global_state.max_block_size.div_ceil(10),
```

Comment:
Maybe this isn't that big a deal, but I think the steady state here is to end up thrashing between increasing and decreasing the block size, so that we are always right around the maximum VID we can process in time, but not hitting it exactly. Not sure if there's a great way to fix that though

Reply:
I considered that, but this oscillation would happen only in a case where we are constantly overwhelmed by transactions.
I have since thought up an easy way to address that: perform this increment only if not only did we serve VID in time and unnecessarily truncated the block, but had some threshold amount of time left over as well.